### PR TITLE
Update Hook 0.1.0

### PR DESCRIPTION
**Pull request by fabiosvm**
Infra update to make Hook 0.1.0 nightly.

### CODE REVIEW

File: bin/yaml/hook.yaml [hunk 1]
```diff
@@ -8,3 +8,9 @@ compilers:
     dir: hook/hook-{name}
     targets:
       - 0.1.0
+    nightly:
+      if: nightly
+      type: nightlytarballs
+      install_always: true
+      targets:
+        - main
```
Comment (mattgodbolt) [inline bin/yaml/hook.yaml:16]: Tested locally and didn't work:
```
2023-01-06 14:22:33,545 lib.ce_install  INFO     compilers/hook/nightly main failed to install: Fetch failure for https://github.com/fabiosvm/hook-lang/releases/download/main/hook-main-linux-x64.tar.gz: <Response [404]>
```

You can repro with:

```
$ bin/ce_install --enable nightly install hook
```

**Comment by fabiosvm** [comment]
Maybe I need to override the `url` field to:

`https://github.com/fabiosvm/hook-lang/archive/refs/heads/{name}.tar.gz`

**Comment by mattgodbolt** [comment]
Still doesn't work:
```
2023-01-06 15:35:46,533 lib.ce_install  INFO     compilers/hook/nightly main failed to install: After unpacking, hook-main-linux-x64 was not a directory
```

**Comment by mattgodbolt** [comment]
(if possible can you please test locally - we set these things up on purpose to be as easy as possible to run locally so I don't end up having to manually test everything)

**Comment by fabiosvm** [comment]
Sure, sure. It's just that I thought there was no way to test locally because I would need to be an AWS. ;)

**Comment by fabiosvm** [comment]
I changed. Now, I'm getting an error:

```
2023-01-06 19:59:20,703 lib.installation_context INFO Fetching https://github.com/fabiosvm/hook-lang/releases/download/0.1.0/hook-0.1.0-linux-x64.tar.gz (357967 bytes)
2023-01-06 19:59:21,420 lib.installation_context INFO 100% of https://github.com/fabiosvm/hook-lang/releases/download/0.1.0/hook-0.1.0-linux-x64.tar .gz
2023-01-06 19:59:21,420 lib.installation_context INFO Piping to tar zxf -
2023-01-06 19:59:21,463 lib.installation_context INFO Moving from staging (/opt/compiler-explorer/staging/6b5e1b27-98ad-4ddb-adf7-32b54df93062/hook-0.1.0-linux-x64) to final destination (/opt/compiler-explorer/hook/hook-0.1.0)
2023-01-06 19:59:21,465 lib.installation_context INFO Destination /opt/compiler-explorer/hook/hook-0.1.0 exists, temporarily moving out of the way (to /opt/compiler-explorer/staging/6b5e1b27- 98ad-4ddb-adf7-32b54df93062/temp_orig)
/opt/compiler-explorer/hook/hook-0.1.0/bin/hook: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /opt/compiler- explorer/hook/hook-0.1.0/bin/hook)
2023-01-06 19:59:21,469 lib.ce_install ERROR compilers/hook 0.1.0 installed OK, but doesn't appear as installed after
Installing compilers/hook/nightly main
2023-01-06 19:59:22,546 lib.installation_context INFO Fetching https://github.com/fabiosvm/hook-lang/archive/refs/heads/main.tar.gz (0 bytes)
2023-01-06 19:59:23,094 lib.installation_context INFO 100% of https://github.com/fabiosvm/hook-lang/archive/refs/heads/main.tar.gz
2023-01-06 19:59:23,095 lib.installation_context INFO Piping to tar zxf -
2023-01-06 19:59:23,179 lib.ce_install INFO compilers/hook/nightly main failed to install: After unpacking, hook-main-linux-x64 was not a directory
Traceback (most recent call last):
   File "/home/f3159258/GitHub/fabiosvm/infra/bin/lib/ce_install.py", line 393, in install
     installable.install()
   File "/home/f3159258/GitHub/fabiosvm/infra/bin/lib/installable/archives.py", line 195, in install
     self.stage(staging)
   File "/home/f3159258/GitHub/fabiosvm/infra/bin/lib/installable/archives.py", line 182, in stage
     raise RuntimeError(f"After unpacking, {self.untar_path} was not a directory")
RuntimeError: After unpacking, hook-main-linux-x64 was not a directory

0 packages installed OK, 0 skipped, and 2 failed installation
Failed:
   compilers/hook 0.1.0
   compilers/hook/nightly main
```

**Comment by mattgodbolt** [comment]
That's the error I see too. It seems like whatever happens when you unpack the tarball it doesn't create a "hook-{name}-linux-x64".  A quick looks:


```
$ curl -sL https://github.com/fabiosvm/hook-lang/archive/refs/heads/main.tar.gz | tar tzvf - | head -1
drwxrwxr-x root/root         0 2023-01-05 23:20 hook-lang-main/
```

looks like the `untar_dir` needs to be changed

**Comment by fabiosvm** [comment]
I can try something else: make `hook.yaml` like this:

```
compilers:
   hook:
     type: tarballs
     compression: gz
     check_exe: bin/hook --version
     url: https://github.com/fabiosvm/hook-lang/releases/download/{name}/hook-{name}-linux-x64.tar.gz
     untar_dir: hook-{name}-linux-x64
     dir: hook/hook-{name}
     targets:
       - 0.1.0
     nightly:
       if: nightly
       type: nightlytarballs
       install_always: true
```

So I get the `GLIBC` error:

```
/opt/compiler-explorer/hook/hook-0.1.0/bin/hook: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /opt/compiler- explorer/hook/hook-0.1.0/bin/hook)
2023-01-06 20:17:07,905 lib.ce_install ERROR compilers/hook 0.1.0 installed OK, but doesn't appear as installed after
0 packages installed OK, 0 skipped, and 1 failed installation
Failed:
   compilers/hook 0.1.0
```

**Comment by mattgodbolt** [comment]
Oh no- that means that somehow whatever has been built has a newer versin of glibc than on your computer. can you run the build normally?

**Comment by fabiosvm** [comment]
I can build `hook` normally. I think this problem is only on my machine.

**Comment by partouf** [comment]
https://github.com/fabiosvm/hook-lang/blob/main/.github/workflows/release.yml#L11 it's being built with ubuntu latest, which is 22.04 since november https://github.blog/changelog/2022-11-09-github-actions-ubuntu-latest-workflows-will-use-ubuntu-22-04/

That won't run on our instances by default, it will need some tweaking.

You could either:
* change the releases to build with `ubuntu-20.04`
* or add a `after_stage_script: patchelf --set-rpath /opt/compiler-explorer/gcc-11.2.0/lib64 bin/hook`
* or a complicated web of setting LD_LIBRARY_PATH here and in CE

**Comment by fabiosvm** [comment]
Thanks @partouf. I changed Hook's build script to build on Ubuntu 20.04.

image

Now installing on `infra` works.

```
Installing compilers/hook 0.1.0
2023-01-07 15:41:03,970 lib.installation_context INFO     Fetching https://github.com/fabiosvm/hook-lang/releases/download/0.1.0/hook-0.1.0-linux-x64.tar.gz (369121 bytes)
2023-01-07 15:41:04,034 lib.installation_context INFO     100% of https://github.com/fabiosvm/hook-lang/releases/download/0.1.0/hook-0.1.0-linux-x64.tar.gz
2023-01-07 15:41:04,034 lib.installation_context INFO     Piping to tar zxf -
2023-01-07 15:41:04,079 lib.installation_context INFO     Moving from staging (/opt/compiler-explorer/staging/c5868dda-a379-45fe-9cfa-1bd863b4ab13/hook-0.1.0-linux-x64) to final destination (/opt/compiler-explorer/hook/hook-0.1.0)
2023-01-07 15:41:04,081 lib.installation_context INFO     Destination /opt/compiler-explorer/hook/hook-0.1.0 exists, temporarily moving out of the way (to /opt/compiler-explorer/staging/c5868dda-a379-45fe-9cfa-1bd863b4ab13/temp_orig)
2023-01-07 15:41:04,086 lib.ce_install  INFO     compilers/hook 0.1.0 installed OK
1 packages installed OK, 0 skipped, and 0 failed installation
```

File: bin/yaml/hook.yaml [hunk 2]
```diff
@@ -8,3 +8,7 @@ compilers:
     dir: hook/hook-{name}
     targets:
       - 0.1.0
+    nightly:
+      if: nightly
+      type: nightlytarballs
+      install_always: true
```
Comment (mattgodbolt) [inline bin/yaml/hook.yaml:14]: there's no `target:` list here so this doesn't do anything. Let me see if I can work out what the name is supposed to be

Comment (mattgodbolt) [reply bin/yaml/hook.yaml:14]: Pinged you in slack: should I update to always install 0.1.0 as the "trunk" build? (we don't usually do this but it's up to you)

**Pull request merged**